sotodlib/hwp/g3thwp.py

```diff
@@ -621,7 +629,134 @@ def write_solution(self, solved, output=None):
+        - locked: bool
+            if non-zero, indicates the HWP is spinning and the position solution is working. \n
+            In this case one should find the hwp_angle populated in the fast data block. \n
+        - hwp_rate: float
+            the "approximate" HWP spin rate, with sign, in revs / second. \n
+            Use placeholder value of 0 for cases when not "locked".
+        - eval: bool
+            if non-zero, the template subtraction is completed.
+        """
+        if self._output is None and output is None:
+            logger.warning('Output file not specified')
+            return
+        if output is not None:
+            self._output = output
+            
+        if len(tod.timestamps) == 0:
+            logger.warning('Input data is empty.')
+            return 
+        
+        start = int(tod.timestamps[0])-self._margin
+        end = int(tod.timestamps[-1])+self._margin
+        try:
+            data = self.load_data(start, end)
+        except Exception as e:
+            logger.error(f"Exception '{e}' thrown while loading HWP data. The specified encoder field is missing.")
+            self._write_empty_solution_h5(tod, output, h5_address)
+            return
+            
+        if len(data) == 0:
+            logger.warning('No HWP data in the specified timestamps.')
+            self._write_empty_solution_h5(tod, output, h5_address)
+            return
+        
+        # calculate HWP angle 
+        logger.debug("analyze")
+        
+        try:
+            solved = self.analyze(data)
+        except Exception as e:
+            logger.error(f"Exception '{e}' thrown while calculating HWP angle. Encoder signal might have too much noise.")
+            self._write_empty_solution_h5(tod, output, h5_address)
+            return 
+            
+        if len(solved['fast_time']) == 0:
+            logger.info('No rotation data in the specified timestamps.')
+            self._write_empty_solution_h5(tod, output, h5_address)
+            return
+            
+        # calculate template subtracted angle
+        try:
+            self.eval_angle(solved)
+        except Exception as e:
+            logger.error(f"Exception '{e}' thrown while the template subtraction")
+            pass
+        
+        # write solution
+        aman = sotodlib.core.AxisManager(tod.dets, tod.samps)
+        aman.wrap_new('timestamps', shape=('samps', ))
+        aman.wrap_new('hwp_angle_ver1', shape=('samps', ))
+        aman.wrap_new('hwp_angle_ver2', shape=('samps', ))
+        aman.wrap_new('stable', shape=('samps', ))
+        aman.wrap_new('locked', shape=('samps', ))
+        aman.wrap_new('hwp_rate', shape=('samps', ))
+        aman.wrap_new('eval', shape=('samps', ))
+        
+        aman.timestamps[:] = tod.timestamps
+        aman.stable[:] = scipy.interpolate.interp1d(solved['slow_time'], solved['stable'], kind='linear', bounds_error=False)(tod.timestamps)
```

Comment:
Don't we want bounds_error=True?  Or is this just a way to make the call faster?

Reply:
"bounds_error=False" is set not to run the extrapolations when the HWP data don't exist at the beginning or the end of the input time range. 
If "bounds_error=True", the process will stop in such cases.
If "bounds_error=False", where is out of input time range will be filled with None, instead of the extrapolation.
I added the error message when the None-filling process is performed.